src/posts/power-of-infrastructure-as-code-iac.md

```diff
@@ -0,0 +1,631 @@
+---
+type: Blog
+title: The power of Infrastructure as Code (IaC)
+subheading: In this post we’ll give an example of how Infrastructure as Code (IaC) is so powerful, by using Terraform to provision some resources in UpCloud.
+authors:
+- mvainio
+tags:
+- Infrastructure as Code
+- Terraform
+date: 2022-06-01
+image: "/blogs/power-of-infrastructure-as-code-iac/power-of-infrastructure-as-code.png"
+featured: true
+
+---
+
+Infrastructure as Code (IaC) can seem intimidating at first glance. Especially when compared with creating cloud resources using a CLI or web console which seems more intuitive. You just choose what you want based on a fixed list of options, and click the button or execute a command. 
+
+Creating cloud resources in this manner is *imperative* as opposed to Infrastructure as Code which is (ideally) *declarative*. With Infrastructure as Code, you define what you want on a high level and it get’s created. Dependencies between different resources, such as load balancer and a virtual machine, are automatically handled by the tool instead of typing out commands or clicking through UI in a correct sequence. 
+
+Our favourite Infrastructure as Code tool at Verifa is Terraform since it is cloud agnostic and is so widely used that it’s easy to find modules and examples. And the best part is, it’s open source! To show the true power of IaC we’re going to provision some resources in [UpCloud]([https://upcloud.com/](https://upcloud.com/)).
+
+## Creating UpCloud resources through the web console
+
+Creating cloud resources using the cloud provider’s web console is a common starting point when testing a new cloud provider, or just starting out in the cloud in general. This method which consists of mostly clicking mouse buttons is sometimes called ClickOps. Let’s create a virtual machine in UpCloud through the [web console]([https://hub.upcloud.com/](https://hub.upcloud.com/)) by clicking on the “Deploy server” button:
```

Comment:
@rubyrat same issue with the link here...

Reply:
done!